src/macros.rs

```diff
@@ -15,6 +15,11 @@ pub fn initial_setup() {
         return;
     }
 
+    unsafe {
+        ocaml_sys::caml_thread_initialize(ocaml_sys::UNIT);
+        ocaml_boxroot_sys::boxroot_setup();
```

Comment:
Note that the call to caml_thread_initialize is already done by boxroot_setup with boxroot 0.4.

Reply:
you could also remove its declaration

Reply:
ah, thank you!